src/spatial/explog.hpp

```diff
@@ -51,7 +52,7 @@ namespace se3
     const Scalar t2 = v.squaredNorm();
     
     const Scalar t = math::sqrt(t2);
-    if(t > 1e-4)
+    if(t > TaylorSeriesExpansion<Scalar>::template precision<4>())
     {
```

Comment:
Minor comment:
Isn't 4 misleading here. The code does a taylor expansion at the 2nd order only...

Reply:
I may adjust the code to go up to order 4th. What do you think?

Reply:
I don't know what is better between a higher order Taylor expansion and a narrower interval of definition of it.
The following code:
```cpp
#include <iostream>
#include <cmath>
#include <iomanip>

int main()
{
    std::cout << std::setprecision(17) << std::fixed;
    std::cout << "sin(t) / t\n";
    for (double d = 0.1; d > 1e-15; d /= 10) {
        double sinc = std::sin(d)/d;
        double sinc_t = 1 - d*d/6;
      std::cout << d << '\t' << sinc << '\t' << sinc_t << '\t' << sinc - sinc_t << std::endl;
    }
    
    std::cout << "\n(1-cos(t)) / t^2\n";
    for (double d = 0.1; d > 1e-15; d /= 10) {
        double cosc = (1-std::cos(d))/(d*d);
        double cosc_t = 0.5 - d*d/24;
      std::cout << d << '\t' << cosc << '\t' << cosc_t << '\t' << cosc - cosc_t << std::endl;
    }
}
```
outputs
```
sin(t) / t

0.10000000000000001	0.99833416646828155	0.99833333333333329	0.00000083313494825
0.01000000000000000	0.99998333341666645	0.99998333333333334	0.00000000008333312
0.00100000000000000	0.99999983333334164	0.99999983333333331	0.00000000000000833
0.00010000000000000	0.99999999833333342	0.99999999833333331	0.00000000000000011
0.00001000000000000	0.99999999998333322	0.99999999998333333	-0.00000000000000011
0.00000100000000000	0.99999999999983336	0.99999999999983336	0.00000000000000000
0.00000010000000000	0.99999999999999833	0.99999999999999833	0.00000000000000000
0.00000001000000000	1.00000000000000000	1.00000000000000000	0.00000000000000000
0.00000000100000000	1.00000000000000000	1.00000000000000000	0.00000000000000000
0.00000000010000000	1.00000000000000000	1.00000000000000000	0.00000000000000000
0.00000000001000000	1.00000000000000000	1.00000000000000000	0.00000000000000000
0.00000000000100000	1.00000000000000000	1.00000000000000000	0.00000000000000000
0.00000000000010000	1.00000000000000000	1.00000000000000000	0.00000000000000000
0.00000000000001000	1.00000000000000000	1.00000000000000000	0.00000000000000000

(1-cos(t)) / t^2

0.10000000000000001	0.49958347219741783	0.49958333333333332	0.00000013886408451
0.01000000000000000	0.49999583334736641	0.49999583333333331	0.00000000001403311
0.00100000000000000	0.49999995832550326	0.49999995833333333	-0.00000000000783007
0.00010000000000000	0.49999999696126451	0.49999999958333335	-0.00000000262206884
0.00001000000000000	0.50000004137018539	0.49999999999583333	0.00000004137435206
0.00000100000000000	0.50004445029117028	0.49999999999995831	0.00004445029121197
0.00000010000000000	0.49960036108132022	0.49999999999999956	-0.00039963891867933
0.00000001000000000	0.00000000000000000	0.50000000000000000	-0.50000000000000000
0.00000000100000000	0.00000000000000000	0.50000000000000000	-0.50000000000000000
0.00000000010000000	0.00000000000000000	0.50000000000000000	-0.50000000000000000
0.00000000001000000	0.00000000000000000	0.50000000000000000	-0.50000000000000000
0.00000000000100000	0.00000000000000000	0.50000000000000000	-0.50000000000000000
0.00000000000010000	0.00000000000000000	0.50000000000000000	-0.50000000000000000
0.00000000000001000	0.00000000000000000	0.50000000000000000	-0.50000000000000000
```

Reply:
I would say here that we are making a Taylor expansion at order 3 instead, and then the error is of order M*|x|^4 / 4! , and this error must be evaluated at the numerical precision of the scalar type.

Reply:
I think the decision should be driven by the numerical precision of the expression we want to evaluate. In this case, the limitation comes from `(1-cos(t)) / t^2`. It seems it starts to be numerically unstable somewhere between `t = 0.001` and `t = 0.0001`. Lets call this limit `T`.

This means that we must find the order `k` of the Taylor expansion such that `M * T^{k+1} / (k+1)!` is close to the machine precision. In this case, `k=3` gives `2e-15` which is good enough. So we need the 3rd order Taylor expansion.

Fortunately, that's what the code does and `TaylorSeriesExpansion<Scalar>::template precision<4>()` is the good value.